input/otlp/logs.go

```diff
@@ -239,3 +245,26 @@ func (c *Consumer) convertLogRecord(
 
 	return event
 }
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStreamField(field string) string {
```

Comment:
Also depending on how performance sensitive this function is, we could also use an ascii table map trick [link](https://go.dev/play/p/NeksqtwNJ0L). It's x7 faster on average than `strings.Map`. It's also way less readable, so I'd say it's not worth it unless this is a hot execution path.